types/profiles/v1/profiles.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2017 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This file deals with the labels, constants, defaults & utility
+// functions related to the profiles
+package v1
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ProvisionPolicyKey type deals with the keys associated with
+// volume provisioning. It is expected that these keys will have
+// corresponding values.
+//
+// USAGE:
+//  When OpenEBS storage is managed with Kubernetes (K8s) as the underlying
+// container orchestrator, then provision policy key can be used as following:
+//
+// apiVersion: storage.k8s.io/v1
+// kind: StorageClass
+// metadata:
+//    name: openebs-jupyter
+//    labels:
+//     provision/replica-count: 2
+//     provision/controller-count: 1
+// provisioner: openebs.io/provisioner-iscsi
+// parameters:
+//   pool: hostdir-var
```

Comment:
@AmitKumarDas can you explain little bit about pool here.

Reply:
The parameters is a map[string]string in K8s storage class. A specific provisioner can pass in different key:value pairs. We have not used pool. We will in future & i should be able to explain it then. 

This also gives me an idea that policies should be within parameters than in labels.

Reply:
yes. i think polices under parameters is a better option. 

but.. can we query for all the storage classes having a certain policy .. in case we put them under parameters?

Reply:
Ha Ha ... You have got me :+1: 

I need to check !!!